device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java

```diff
@@ -552,41 +547,6 @@ else if (value == null)
             throw new UnsupportedOperationException("Must re-register this client to a multiplexing client before using it");
         }
 
-        switch (optionName)
```

Comment:
Now that we've gotten rid of the transportClient paths, there is less logic needed in the DeviceClient class and we can just rely on the common implementation of this method in ModuleClient.